lua/auto-save/utils/colors.lua

```diff
@@ -44,27 +56,42 @@ function M.get_hl(name)
   return hl
 end
 
----@param fg string forecrust color
+---@param fg string foreground color
 ---@param bg string background color
 ---@param alpha number number between 0 and 1. 0 results in bg, 1 results in fg
-function M.blend(fg, bg, alpha)
-  bg = hex_to_rgb(bg)
-  fg = hex_to_rgb(fg)
+local function blend(fg, bg, alpha)
+  local bg_hex = hex_to_rgb(bg)
+  local fg_hex = hex_to_rgb(fg)
 
   local blendChannel = function(i)
-    local ret = (alpha * fg[i] + ((1 - alpha) * bg[i]))
+    local ret = (alpha * fg_hex[i] + ((1 - alpha) * bg_hex[i]))
     return math.floor(math.min(math.max(0, ret), 255) + 0.5)
   end
 
   return string.format("#%02X%02X%02X", blendChannel(1), blendChannel(2), blendChannel(3))
 end
 
-function M.darken(hex, amount, bg)
-  return M.blend(hex, bg or M.bg, math.abs(amount))
+--- @param dim_value number
+M.apply_colors = function(dim_value)
+    if dim_value > 0 then
+      MSG_AREA = get_hl("MsgArea")
+      if MSG_AREA.foreground ~= nil then
+        MSG_AREA.background = (MSG_AREA.background or get_hl("Normal")["background"])
+        -- TODO maybe there is a hiccup here, test a bit
```

Comment:
Is this TODO still relevant?

Reply:
I don't understand what the original author did here. In my opinion the naming of the variables is incorrect, we are somewhere mixing up foreground and background. Here or in the `blend` function.

But it works, so I don't want to touch it further ;-)

Reply:
Fair enough. TBH I'm not sure if dimming should be even implemented here, it should be the responsibility of some UI plugin. IMO we should deprecate it and come up with a bit more general solution.

Up to you if you want to update the TODO with a bit more details or even a deprecation notice.